Source/SA/RFmx Instr/RFmxInstr.cs

```diff
@@ -0,0 +1,85 @@
+                        LocalOscillatorConfiguration lo2Config = LocalOscillatorConfiguration.GetDefault();
+                        lo2Config.ChannelName = "LO2";
+                        instrConfig.LoConfigurations = new LocalOscillatorConfiguration[] { lo1Config, lo2Config };
+                        break;
+                }
+                return instrConfig;
+            }
+        }
+        #endregion
+
+        #region Instrument Configurations
+        public static void ConfigureInstrument(RFmxInstrMX instrHandle, InstrumentConfiguration instrConfig)
+        {
+            foreach (LocalOscillatorConfiguration loConfig in instrConfig.LoConfigurations)
+            {
+                switch (loConfig.SharingMode)
+                {
+                    case LocalOscillatorSharingMode.None:
+                        instrHandle.SetAutomaticSGSASharedLO(loConfig.ChannelName, RFmxInstrMXAutomaticSGSASharedLO.Disabled);
+                        instrHandle.SetLOExportEnabled(loConfig.ChannelName, false);
+                        instrHandle.SetLOSource(loConfig.ChannelName, RFmxInstrMXConstants.LOSourceOnboard);
+                        break;
+                    case LocalOscillatorSharingMode.Manual:
+                        instrHandle.SetAutomaticSGSASharedLO(loConfig.ChannelName, RFmxInstrMXAutomaticSGSASharedLO.Disabled);
+                        instrHandle.SetLOExportEnabled(loConfig.ChannelName, loConfig.ExportEnabled);
+                        instrHandle.SetLOSource(loConfig.ChannelName, loConfig.Source);
+                        break;
+                    default: // default to automatic case
+                        instrHandle.SetAutomaticSGSASharedLO(loConfig.ChannelName, RFmxInstrMXAutomaticSGSASharedLO.Enabled);
+                        instrHandle.ResetAttribute(loConfig.ChannelName, RFmxInstrMXPropertyId.LOExportEnabled);
+                        instrHandle.ResetAttribute(loConfig.ChannelName, RFmxInstrMXPropertyId.LOSource);
+                        break;
+                }
+
+                switch (loConfig.OffsetMode)
+                {
+                    case LocalOscillatorOffsetMode.NoOffset:
+                        instrHandle.SetLOLeakageAvoidanceEnabled(loConfig.ChannelName, RFmxInstrMXLOLeakageAvoidanceEnabled.False);
+                        instrHandle.SetDownconverterFrequencyOffset(loConfig.ChannelName, 0.0);
+                        break;
+                    case LocalOscillatorOffsetMode.UserDefined:
+                        instrHandle.SetLOLeakageAvoidanceEnabled(loConfig.ChannelName, RFmxInstrMXLOLeakageAvoidanceEnabled.False);
+                        instrHandle.SetDownconverterFrequencyOffset(loConfig.ChannelName, loConfig.Offset_Hz);
+                        break;
+                    default: // default to automatic case
+                        instrHandle.SetLOLeakageAvoidanceEnabled(loConfig.ChannelName, RFmxInstrMXLOLeakageAvoidanceEnabled.True);

```

Comment:
I am not sure on that but on the driver level resetting the attribute on SG and SA would mean that we switch back to auto offset mode. Therefore my question is if it was sufficient to no use the LoLeakageAvoidance at all?

Reply:
we could reset the property rather than setting it if that makes more sense here?